### ISSUE TITLE
Sonarr losing Plex List Authentication after certain time period

### ISSUE THREAD

**Issue by RileyXX** [Sonarr/Sonarr#5154]
Every once in a while my sonarr and radarr server settings I have to go back and manually re-authenticate my Plex Watchlist. Is there anyway to automate this or stop it from making me re-authenticate in the first place? It sounds like whatever API sonarr is using with Plex the tokens expire after a certain amount of time. I can't tell if it's 30 or 60 days but probably one of the two.

Edit: Also posted on Reddit https://www.reddit.com/r/sonarr/comments/xdy3f3/sonarr_losing_plex_list_authentication_after/

**Comment by RileyXX** [comment]
Currently I'm running a script called [PlexTraktSync](https://github.com/Taxel/PlexTraktSync) that syncs my Plex watch history and watchlist. And then I have a public watchlist on Trakt.tv that I use for auto importing all my watchlist series from Plex and Trakt into sonarr. So technically I don't need to setup Plex lists in sonarr but still it would be nice if it was still setup and synced without needing me to re-authenticate it every month or two.

**Comment by markus101** [comment]
Are you certain the token expired on it's own and it wasn't expired after resetting your password recently (and if following Plex's recommendation, signing out of all devices). We're using the exact same mechanism we use for Plex Media Server authentication and had no reported issues (and it's something I've used consistently as well).

**Comment by markus101** [comment]
Please update your issue to include all the information requested information when opening issues instead of blowing away the issue entirely.

**Comment by RileyXX** [comment]
> 

Most of the template was unnecessary. Yes the auth token expired on it's own according to the logs. Same thing for both the Plex "Lists" page and Plex "connections" page in sonarr. Both expired tokens. Seems like a limitation with the Plex api. 
`[v3.0.9.1555] NzbDrone.Core.Notifications.Plex.PlexAuthenticationException: Unauthorized - AuthToken is invalid`

I know some api's will let you renew the tokens automatically but I couldn't find any documentation online for the lifetime of Plex API tokens. According to the community [Plex-Api on Github](https://github.com/jc21/plex-api/blob/master/docs/Documentation.md#plexapi) "_There is no formal documentation referencing the lifespan of a token. These appear to have a long shelf-life, but it is possible they rotate occasionally_."

**Comment by markus101** [comment]
There is a reason we created the template and ask that it's filled out, we can decide what is relevant or not.

The error indicates that the token is expired, but doesn't explain how it became expired, after some additional digging I've found that tokens are expired based on last used time (if they're not used for a period of time they become invalid). Unfortunately neither of the actions we're performing trigger that they're in use, though there is another end point we can use to validate the token which in turn updates it's last seen status and keeps it valid.
